<?xml version="1.0" ?>

<launch> 
  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="robot_name" default="thormang3" />
  <arg name="pinned" default="false"/>
  <arg name="use_present" default="true" />
  <arg name="hand_type_0" default="no_hand" />  
  <arg name="hand_type_1" default="no_hand" />
    
  <group ns="/$(arg robot_name)">    
    <!-- Load the URDF into the ROS Parameter Server -->
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find thormang3_description)/urdf/thormang3.xacro' robot_name:=$(arg robot_name) pinned:=$(arg pinned) hand_type_0:=$(arg hand_type_0) hand_type_1:=$(arg hand_type_1)" />
    
    <!-- send joint values -->
    <node name="robotis_joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="use_gui" value="FALSE"/>
        <param name="rate" value="30"/>
        <remap from="/joint_states" to="/$(arg robot_name)/robotis/thormang3/joint_states" />
        <rosparam param="source_list" if="$(arg use_present)">["robotis/thormang3/present_joint_states"]</rosparam>
        <rosparam param="source_list" unless="$(arg use_present)">["robotis/thormang3/goal_joint_states"]</rosparam>
    </node>

    <!-- Combine joint values to TF-->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <remap from="/joint_states" to="/$(arg robot_name)/robotis/thormang3/joint_states"/>
    </node>

    <!-- Gripper Pub-->
    <node name="thormang3_rviz_pub" pkg="thormang3_description" type="thormang3_rviz_pub">
    </node>
  </group>
</launch>
